docs/extenders/Error Handling.md

```diff
@@ -0,0 +1,84 @@
+# Error Handling Best Practices
+
+This page contains best practices in handling errors for Zowe Explorer extenders.
+
+## Contents
+
+  - [Objectives](#objectives)
+  - [Error Message Format](#error-message-format)
+  - [Logging of Error Message](#logging-of-error-message)
+  - [Showing Errors to Users](#showing-errors-to-users)
+
+## Objectives
+
+By following these best practices, Zowe Explorer extenders will be able to provide a meaningful error message and help users identify the root cause of a problem quickly and easily.
+
+We encourage extenders to use this guidance in order to:
+
+- Help users understand which extension is causing an error, by including the name of the extension in the error message.
+- Give developers more information in error logs.
+- Standardize the error message format across all of Zowe Explorer's extenders.
+
+## Error Message Format
+
+We highly recommend the usage of this error message format:
+
+```
+Extension Name Error - Error Message
+```
+
+Ex. for the Zowe Explorer FTP extension, an error message might look like:
+
+```
+Zowe Explorer FTP Extension Error - Unable to delete node…
+```
+
+## Logging of Error Message
+
+We highly recommend the usage of `imperative.Logger` when logging an extender's error messages.
+
+These logs have a stardard format that is used by Zowe Explorer and will be consistent across Zowe components.
+
+Example:
+
+```
+public constructor(protected log: imperative.Logger) {}
+  try {
+  ...
+  } catch (error) {
+      this.log.error(error);
+  }
+```
+
+## Showing Errors to Users
+
+We highly recommend using VSCode's API to show messages to users. This will provide a standard format
+that is similar to Zowe Explorer and VSCode.
+
+- For showing information to the user: [vscode.window.showInformationMessage](https://code.visualstudio.com/api/references/vscode-api#window.showInformationMessage)
+
+```
+vscode.window.showInformationMessage(
+                localize("findUSSItem.unsuccessful", "File does not exist. It
+                may have been deleted.")
+            );
+```
+
+- For showing errors to the user: [vscode.window.showErrorMessage](https://code.visualstudio.com/api/references/vscode-api#window.showErrorMessage)
+
+```
+vscode.window.showErrorMessage(
+                localize("deactivate.error", "Unable to delete temporary
+                folder.") + err
+            );
+```
```

Comment:
For the `vscode.window.show*Message` code samples, I think they will be easier for extenders to understand if we simplify them to just the essentials. Although localization is important, the `localize` function in these examples can be a bit distracting. Unless we want to link the extenders to documentation on localizing, perhaps we can consider removing the `localize` function out of the example, and just pass in a simple error/information string instead.